python/ray/data/context.py

```diff
@@ -216,6 +220,8 @@ class DataContext:
             always written to the Ray Data log file.
         print_on_execution_start: If ``True``, print execution information when
             execution starts.
+        s3_try_create_dir: If ``True``, try to create directories on S3 when a write
+            call is made with a S3 URI.
```

Comment:
Should this go in `DataContext` directly or `ExecutionOptions` (not suggesting one way or another, just genuinely not sure about how we decide)

Reply:
Not sure if I have the context to know which would be preferable for this, will leave unresolved if anyone has strong feelings please weight in.